modules/fedimint-wallet/src/lib.rs

```diff
@@ -1536,7 +1536,13 @@ pub fn is_address_valid_for_network(address: &Address, network: Network) -> bool
 #[instrument(level = "debug", skip_all)]
 pub async fn run_broadcast_pending_tx(db: Database, rpc: DynBitcoindRpc, tg_handle: &TaskHandle) {
     while !tg_handle.is_shutting_down() {
-        broadcast_pending_tx(db.begin_transaction().await, &rpc).await;
+        broadcast_pending_tx(
+            db.begin_transaction()
+                .await
+                .with_module_prefix(LEGACY_HARDCODED_INSTANCE_ID_WALLET),
```

Comment:
Needs to be fixed. I'm ok with a follow-up, but outside tests and the old client we should not use legacy ids.

Reply:
Hmm yes I think this is best as a follow up. I think the issue here is that, in order to not use the hardcoded constant, we'd need to pass the module_instance_id all the way down from ModuleGen::init. We can't create a database transaction during ModuleGen::init because this thread is polling the DB and needs to create a new transaction every time. Passing it down all the way through ModuleGen::init though means individual modules are now aware of their ID, which I think we're trying to avoid. Maybe this would also be a good candidate for the notification work @Maan2003 

@dpc is there an existing issue for removing the legacy harded instance ids or should I make a new one?

Reply:
Wait, what? This is a module code already. Whatever this module is holding in that `db: Database` should be already isolated.

Reply:
If we need to pass modules a whole db handle, we might need a similar wrapper for the whole `Database`

Reply:
The module must never get it's unprivileged hands on anything remotely resembling unisolated access to the database. :D